platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -307,6 +273,75 @@ class ConfigurationCacheEncryptionIntegrationTest extends AbstractConfigurationC
         configurationCache.assertStateStored()
     }
 
+    def "encryption service honors default keystore type"() {
+        def defaultKeystoreType = KeyStore.defaultType
+        buildFile """
+        def encryptionService = services.get(org.gradle.internal.encryption.EncryptionConfiguration)
+        // ensure encryption service is triggered
+        assert encryptionService.encrypting
+        """
+        when:
+        run("help", "-i")
+
+        then:
+        outputContains("Encryption key source: default Gradle configuration cache keystore (${defaultKeystoreType})")
+    }
+
+    def "encryption service honors explicitly defined keystore type"() {
```

Comment:
❓ Do we have a test that validates different ways of setting the keystore and their priority? This test only checks that the internal option has priority over `keystore.type` property

The more extended test probably lands itself well as a data-driven test. Though, as noted above, I wonder if a unit test would be enough here?

What concerns me as well is that we would explicitly reject the `dks` default, but sort of allow it if provided via the internal option. It seems there are tricky invariants in the keystore type selection that we are not explicitly testing yet.

Reply:
🤔  I just realized that the `EncryptionService` is no longer tied to CC in its operations and where it's located. We should make sure then we don't mention CC in info messages and **we should probably move the tests** that are not about CC into a separate place.

Reply:
I added an integration test that validates the fallback behavior